types/ramda/index.d.ts

```diff
@@ -1005,6 +1006,45 @@ declare namespace R {
         mergeAll<T>(list: any[]): T;
 
         /**
+         * Creates a new object with the own properties of the first object merged with the own properties of the second object.
+         * If a key exists in both objects:
+         * and both values are objects, the two values will be recursively merged
+         * otherwise the value from the first object will be used.
+         */
+        mergeDeepLeft<T1, T2>(a: T1, b: T2): T1 & T2;
+        mergeDeepLeft<T1>(a: T1): <T2>(b: T2) => T1 & T2;
+
+        /**
+         * Creates a new object with the own properties of the first object merged with the own properties of the second object.
+         * If a key exists in both objects:
+         * and both values are objects, the two values will be recursively merged
+         * otherwise the value from the second object will be used.
+         */
+        mergeDeepRight<A, B>(a: A, b: B): A & B;
+        mergeDeepRight<A>(a: A): <B>(b: B) => A & B;
+
+        /**
+         * Creates a new object with the own properties of the two provided objects. If a key exists in both objects:
+         * and both associated values are also objects then the values will be recursively merged.
+         * otherwise the provided function is applied to associated values using the resulting value as the new value
+         * associated with the key. If a key only exists in one object, the value will be associated with the key of the resulting object.
+         */
+        mergeDeepWith<T1, T2>(fn: (x: any, z: any) => any, a: T1, b: T2): T1 & T2;
```

Comment:
This type isn't quite right -- if `fn` is allowed to return anything, then the output might not have the same type as the inputs. Should `fn` be `<T>(a: T, b: T) => T`?